src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_emitter.cpp

```diff
@@ -226,9 +193,13 @@ void jit_brgemm_emitter::emit_brgemm_kernel_call(const brgemm_kernel_t *brg_kern
     h->lea(h->r10, h->ptr[abi_param1 + GET_OFF(amx_tile_config)]);
     h->mov(h->qword[h->rsp + GET_OFF_BRGEMM_ARGS(amx_tile_config)], h->r10);
 
-    h->mov(abi_param1, reinterpret_cast<uintptr_t>(brg_kernel));
-    h->mov(abi_param2, reinterpret_cast<uintptr_t>(&m_ctx));
-    h->mov(abi_param3, h->rsp);
+    h->mov(abi_param1, reinterpret_cast<uintptr_t>(m_kernel_executor.get()));
+    h->mov(abi_param2, h->rsp);
+#ifdef _WIN32
+    // Before function call we should also allocate stack area for ABI parameters (shadow space)
+    reserved_stack_size += 32;
+    h->sub(h->rsp, 32);
+#endif
```

Comment:
👍  Should we also fix `TppEmitter::emit_impl` in the same way? In addition, we could modify a bit `jit_brgemm_copy_b_emitter::emit_kernel_call`: shadow space is taken into account there, but it is "hidden" under the number of params passed on stack -- as a result, the code seems a bit confusing.

And since this code must be always called before each function, maybe we can move it to some helper? As an option, we could move it to `internal_call_rsp_align();` (and the opposite operation to `internal_call_rsp_restore()`), or create a new one, which will call `internal_call_rsp_align` (it can become private) and perform Windows related rsp manipulations if needed. What do you think?

Reply:
TppEmitter fixed, thanks.
I agree, let's move shadow space handling to `internal_call_rsp_align()`, since currently it's done in every binary call emitter manually, which is very error-prone.

The only concern here, is that we will have to do extra `rsp` increment/decrement if additional regs (after the first 4) are passed on the stack. However, I don't think this is a big problem, especially considering the fact that this is happening only in `jit_brgemm_copy_b_emitter`, and would be eliminated when we create executor for this emitter.

@a-sidorova, what are your thoughts on this?

Reply:
To move shadow space handling to `internal_call_rsp_align()` is really good idea! And I think that to make additional increment/decrement for `rsp` doesn't matter because these instruction calls are really light. Especially in comparison with kernel in `jit_brgemm_copy_b_emitter`. Also if there is another kernel with `>4` args, I think it will be more complex and heavier anyway than `add/sub` with `rsp.

Thus, I don't have any concerns here.

Reply:
Perfect! I don't have any concerns regarding performance, too